src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,105 @@
-export const PeopleFilters = () => {
+import { Link, useSearchParams } from 'react-router-dom';
+import { Person } from '../types/Person';
+import React, { useState, useEffect } from 'react';
+
+interface PeopleFiltersProps {
+  persons: Person[];
+  onFilterChange: (filter: 'all' | 'male' | 'female') => void;
+
+  onClick: (century: string | 'all' | null) => void;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const PeopleFilters: React.FC<PeopleFiltersProps> = ({
+  onFilterChange,
+  onClick,
+  onChange,
+}) => {
+  const [filter, setFilter] = useState<'all' | 'male' | 'female'>('all');
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [searchParams, setSearchParams] = useSearchParams();
+  const centuryQuery = searchParams.getAll('century') || [];
+
+  useEffect(() => {
+    const currentFilter = searchParams.get('filter') as
+      | 'all'
+      | 'male'
+      | 'female';
+    const currentSearch = searchParams.get('search') || '';
+
+    if (currentFilter) {
+      setFilter(currentFilter);
+    } else {
+      setFilter('all');
+    }
+
+    if (centuryQuery.length > 0) {
+      searchParams.set('century', centuryQuery.join(','));
+    } else {
+      searchParams.delete('century');
+    }
```

Comment:
This logic seems redundant because `centuryQuery` is already an array. You might want to ensure that `centuryQuery` is correctly set from the URL parameters initially.